### PR TITLE
Parametrize probes

### PR DESCRIPTION
**Pull request by mrsabath**
This PR addresses #307 by parametrizing Probes and moving them to values.yaml

### CODE REVIEW

File: charts/spire/README.md [hunk 1]
```diff
@@ -292,6 +292,10 @@ Kubernetes: `>=1.21.0-0`
 | spire-server.controllerManager.image.repository | string | `"spiffe/spire-controller-manager"` | The repository within the registry |
 | spire-server.controllerManager.image.tag | string | `"0.2.2"` | Overrides the image tag |
 | spire-server.controllerManager.image.version | string | `""` | This value is deprecated in favor of tag. (Will be removed in a future release) |
+| spire-server.controllerManager.livenessProbe.httpGet.path | string | `"/healthz"` |  |
```
Comment (kfox1111) [inline charts/spire/README.md:295]: This whole pr looks very nice. Thanks. :)

The one thing I think we may want to do/not do is to make the paths/ports/scheme overridable. there isn't a different setting the user would ever need to set for those? so maybe we hard code those still.

Comment (kfox1111) [reply charts/spire/README.md:295]: And maybe make the rest of the missing liveness/readyness/startup probe settings configurable?

Comment (mrsabath) [reply charts/spire/README.md:295]: So @kfox1111 , are you suggesting format something like this?:

```
          readinessProbe:
            httpGet:
              scheme: HTTP
              path: /ready
              port: healthz
            failureThreshold: {{ .Values.startupProbe.failureThreshold }}
            initialDelaySeconds: {{ .Values.startupProbe.initialDelaySeconds }}
            periodSeconds: {{ .Values.startupProbe.periodSeconds }}
            successThreshold: {{ .Values.startupProbe.successThreshold }}
            timeoutSeconds: {{ .Values.startupProbe.timeoutSeconds }}
 ```

Comment (kfox1111) [reply charts/spire/README.md:295]: yeah. that kind of thing.

Comment (kfox1111) [reply charts/spire/README.md:295]: That way, the user can configure the settings that may differ cluster to cluster (slower cluster, higher timeouts) but not have to worry about the stuff that is internal to the pod that won't change on them.

**Review by marcofranssen — APPROVED** [review]
LGTM 🚀

**Review by kfox1111 — APPROVED** [review]
LGTM

**Pull request merged**